srcCpp/RTIDDSImpl.cxx

```diff
@@ -2723,15 +2270,17 @@ unsigned long RTIDDSImpl<T>::GetInitializationSampleCount()
      */
     initializeSampleCount = (std::max)(
             initializeSampleCount,
-            (unsigned long) _SendQueueSize);
+            (unsigned long) _PM->get<int>("sendQueueSize"));
 
     /*
      * If we are using batching we need to take into account tha the Send Queue
      * will be per-batch, therefore for the number of samples:
      */
-    if (_BatchSize > 0) {
+    if (_PM->get<long>("batchSize") > 0) {
         initializeSampleCount = (std::max)(
-                _SendQueueSize * (_BatchSize / _DataLen),
+                _PM->get<int>("sendQueueSize") *
+                        (_PM->get<long>("batchSize") /
+                        _PM->get<unsigned long>("dataLen")),
```

Comment:
This need to be changed as well as RawTransport Branch.

But if we change it to 
`_PM->get<unsigned long long>("dataLen")),`

Compiling with gcc 7.3.0 I got the following error:
```
RTIDDSImpl.cxx: In member function ‘long unsigned int RTIDDSImpl<T>::GetInitializationSampleCount()’:
RTIDDSImpl.cxx:2284:38: error: no matching function for call to ‘max(long long unsigned int, long unsigned int&)’
                 initializeSampleCount);
```
We need to add a new casting for **initializeSampleCount**  
So, in the end, this needs to be like:
```
        initializeSampleCount = (std::max)(
                _PM->get<int>("sendQueueSize") *
                        (_PM->get<long>("batchSize") /
                        _PM->get<unsigned long long>("dataLen")),
                (unsigned long long) initializeSampleCount);
```

Reply:
Fixed